ICSharpCode.Decompiler/IL/Transforms/UsingTransform.cs

```diff
@@ -381,10 +383,16 @@ bool ImplementsForeachPattern(IType type)
 			}
 		}
 
-		bool MatchNullCheckOrTypeCheck(ILInstruction condition, ref ILVariable objVar, KnownTypeCode disposeType)
+		bool MatchNullCheckOrTypeCheck(ILInstruction condition, ref ILVariable objVar, KnownTypeCode disposeType, out bool isInlinedIsInst)
```

Comment:
is `isInlinedIsInst` really necessary?

Reply:
This additional boolean value was added in order to make the pattern matching more strict. I did this mostly because I do not really know how strict the ILSpy pattern-matching code should be in this component.